packages/rs-dpp/src/data_contract/document_type/property/mod.rs

```diff
@@ -382,19 +383,19 @@ impl DocumentPropertyType {
         }
     }
 
+    /// Reads an optional value from the buffer
+    /// Returns an optional value, as well as a boolean to indicate if we have finished the buffer
     pub fn read_optionally_from(
         &self,
         buf: &mut BufReader<&[u8]>,
         required: bool,
-    ) -> Result<Option<Value>, ProtocolError> {
+    ) -> Result<(Option<Value>, bool), ProtocolError> {
```

Comment:
Wouldn't be better to use Cursor to figure out if there is still something there?

Reply:
I think this is more efficient, since we have already looked at memory.

Reply:
Where did we look? This code looks much more complex than using Cursor that supposed to be used in such cases. You can easily check if something has not been read yet

Reply:
We agreed that I'll make a PR to fix it

Reply:
I did not agree that that was the right solution, but I said that if you strongly believe that you want to do that you can and we can then weigh the pros and the cons of the change.

Reply:
Cursor won't help in this case since we are dealing with inner objects: we have nothing to read for this object but we still have data for parent one